### PR TITLE
#1539 Avatar upload backend

### LINKED ISSUE #1539: Avatar upload backend
Backend ticket split from #715

### PR DESCRIPTION
**Pull request by mannykary**
### What does this pull request aim to achieve? 💯

Backend for avatar upload. Added two endpoints for org and user avatar upload, and a module for uploading files to S3 CDN. Also added Localstack container to mock AWS resources locally.

Closes #1539 

- [X] I have checked that no one else is working on similar changes.
- [X] I have read the latest [**README**](README.md) and understand the development workflow.
- [X] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix. The branch is created in a cloned version of this repo, **not a forked repo**.
- [n/a] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix). _This is being merged into the `feature/1405` branch, which is the frontend branch for Avatar Upload. We will merge the latest from staging into that branch._
- [X] There are no merge conflicts.
- [X] There are no console warnings and errors.
- [X] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: backend/lib/endpoints/schema/organisations.js [hunk 1]
```diff
@@ -40,6 +40,13 @@ const createOrganisationSchema = {
     .required(["location"]),
 };
 
+const createOrganisationAvatarSchema = {
+  params: strictSchema().prop("organisationId", S.string().required()),
+  body: strictSchema()
+    .prop("file", S.required())
+    .prop("userId", S.required()),
```
Comment (joshmorel) [inline backend/lib/endpoints/schema/organisations.js]:
```suggestion
  body: strictSchema().prop("file", S.required()),
```

The `userId` is attached to req via a decorator in the auth.js plugin so we don't need to include it in the body.

**Review by joshmorel — CHANGES_REQUESTED** [review]
Thanks @mannykary . Except for the one minor suggestion this seems good. I was able to get it working locally with postman.

I guess the next step is for @ashminbhandari to merge and get it all working together with the frontend then we can see it in real action in a review build?

The only other thing I guess are going to allow a user to delete their avatar?

File: backend/lib/endpoints/schema/organisations.js (hunk 1 above)
Comment (ashminbhandari) [reply backend/lib/endpoints/schema/organisations.js]: I am working on the uploads ATM. Thanks.

Comment (mannykary) [reply backend/lib/endpoints/schema/organisations.js]: Ah good catch, I'll remove it from here.

**Comment by mannykary** [comment]
> The only other thing I guess are going to allow a user to delete their avatar?

I'll need to check the requirements but we probably should allow a user to delete their avatar so that it is removed from S3 as well as their Mongo object.

Edit: I just check the [spec](https://www.notion.so/fightpandemics/Avatars-for-Profiles-e4af32c040c94035a041c76b3c9a1e97) and there is no requirement for deleting avatar photos, but I feel like there should be one. I'm going to assume that it's outside the scope for now, and ask Krithiga to see if we should add that functionality in a separate ticket.

File: backend/lib/endpoints/schema/organisations.js (hunk 1 above)
Comment (mannykary) [reply backend/lib/endpoints/schema/organisations.js]: @joshmorel Resolved in this [commit](https://github.com/FightPandemics/FightPandemics/pull/1540/commits/b66fab6f892da514fd649f8cf9f5aaf27c382601).

**Comment by joshmorel** [comment]
> I'll need to check the requirements but we probably should allow a user to delete their avatar so that it is removed from S3 as well as their Mongo object.
> 
> Edit: I just check the [spec](https://www.notion.so/fightpandemics/Avatars-for-Profiles-e4af32c040c94035a041c76b3c9a1e97) and there is no requirement for deleting avatar photos, but I feel like there should be one. I'm going to assume that it's outside the scope for now, and ask Krithiga to see if we should add that functionality in a separate ticket.

Sounds good. I think we can still go forward & merge this into #1405 so @ashminbhandari can continue with FE/BE integration while we clarify this requirement.

**Review by joshmorel — APPROVED** [review]

**Comment by Monica0843** [comment]
Issue: 
Where choosing a file over 5MB, there is no warning message or anything alike. The file window just gets dismissed. 
Environment: OS:  MacOS, browser: Chrome

**Pull request merged**